test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
-@test dA * Diagonal(b) == scale!(copy(sA), b)
-b = randn(3)
-@test Diagonal(b) * dA == Diagonal(b) * sA
-@test Diagonal(b) * dA == scale!(sC, b, sA)
-@test Diagonal(b) * dA == scale!(b, copy(sA))
-
-@test dA * 0.5            == sA * 0.5
-@test dA * 0.5            == scale!(sC, sA, 0.5)
-@test dA * 0.5            == scale!(copy(sA), 0.5)
-@test 0.5 * dA            == 0.5 * sA
-@test 0.5 * dA            == scale!(sC, sA, 0.5)
-@test 0.5 * dA            == scale!(0.5, copy(sA))
-@test scale!(sC, 0.5, sA) == scale!(sC, sA, 0.5)
-
-# copy!
-let
-    A = sprand(5, 5, 0.2)
-    B = sprand(5, 5, 0.2)
-    copy!(A, B)
-    @test A == B
-    @test pointer(A.nzval) != pointer(B.nzval)
-    @test pointer(A.rowval) != pointer(B.rowval)
-    @test pointer(A.colptr) != pointer(B.colptr)
-    # Test size(A) != size(B), but length(A) == length(B)
-    B = sprand(25, 1, 0.2)
-    copy!(A, B)
-    @test A[:] == B[:]
-    # Test various size(A) / size(B) combinations
-    for mA in [5, 10, 20], nA in [5, 10, 20], mB in [5, 10, 20], nB in [5, 10, 20]
-        A = sprand(mA,nA,0.4)
-        Aorig = copy(A)
-        B = sprand(mB,nB,0.4)
-        if mA*nA >= mB*nB
-            copy!(A,B)
-            @assert(A[1:length(B)] == B[:])
-            @assert(A[length(B)+1:end] == Aorig[length(B)+1:end])
-        else
-            @test_throws BoundsError copy!(A,B)
+
+@testset "Scale and scale!" begin
+    @test dA * Diagonal(b) == sA * Diagonal(b)
+    @test dA * Diagonal(b) == scale!(sC, sA, b)
+    @test dA * Diagonal(b) == scale!(copy(sA), b)
+    b = randn(3)
+    @test Diagonal(b) * dA == Diagonal(b) * sA
+    @test Diagonal(b) * dA == scale!(sC, b, sA)
+    @test Diagonal(b) * dA == scale!(b, copy(sA))
+
+    @test dA * 0.5            == sA * 0.5
+    @test dA * 0.5            == scale!(sC, sA, 0.5)
+    @test dA * 0.5            == scale!(copy(sA), 0.5)
+    @test 0.5 * dA            == 0.5 * sA
+    @test 0.5 * dA            == scale!(sC, sA, 0.5)
+    @test 0.5 * dA            == scale!(0.5, copy(sA))
+    @test scale!(sC, 0.5, sA) == scale!(sC, sA, 0.5)
+end
+
+@testset "copy!" begin
+    let
+        A = sprand(5, 5, 0.2)
+        B = sprand(5, 5, 0.2)
+        copy!(A, B)
+        @test A == B
+        @test pointer(A.nzval) != pointer(B.nzval)
+        @test pointer(A.rowval) != pointer(B.rowval)
+        @test pointer(A.colptr) != pointer(B.colptr)
+        # Test size(A) != size(B), but length(A) == length(B)
+        B = sprand(25, 1, 0.2)
+        copy!(A, B)
+        @test A[:] == B[:]
+        # Test various size(A) / size(B) combinations
+        for mA in [5, 10, 20], nA in [5, 10, 20], mB in [5, 10, 20], nB in [5, 10, 20]
+            A = sprand(mA,nA,0.4)
+            Aorig = copy(A)
+            B = sprand(mB,nB,0.4)
+            if mA*nA >= mB*nB
+                copy!(A,B)
+                @assert(A[1:length(B)] == B[:])
+                @assert(A[length(B)+1:end] == Aorig[length(B)+1:end])
+            else
```

Comment:
Have we guidelines for testset descriptions? In this case, why capitalize every first letter?